lib/bloc/account/account_bloc.dart

```diff
@@ -331,4 +332,45 @@ class AccountBloc extends Cubit<AccountState> with HydratedMixin {
     _log.info("mnemonicsValidated");
     emit(state.copyWith(verificationStatus: VerificationStatus.VERIFIED));
   }
+
+  List<PaymentMinutiae> filterPaymentList() {
+    final nonFilteredPayments = state.payments;
+    final paymentFilters = state.paymentFilters;
+
+    var filteredPayments = nonFilteredPayments;
+    // Apply date filters, if there's any
+    if (paymentFilters.fromTimestamp != null || paymentFilters.toTimestamp != null) {
+      filteredPayments = nonFilteredPayments.where((paymentMinutiae) {
+        final fromTimestamp = paymentFilters.fromTimestamp;
+        final toTimestamp = paymentFilters.toTimestamp;
+        final milliseconds = paymentMinutiae.paymentTime.millisecondsSinceEpoch;
+        if (fromTimestamp != null && toTimestamp != null) {
+          return fromTimestamp < milliseconds && milliseconds < toTimestamp;
+        }
+        return true;
+      }).toList();
+    }
+
+    // Apply payment type filters, if there's any
+    if ((paymentFilters.filters != null && paymentFilters.filters != sdk.PaymentTypeFilter.values)) {
+      filteredPayments = filteredPayments.where((paymentMinutiae) {
+        for (var f in paymentFilters.filters!) {
```

Comment:
paymentFilters.filters can be extracted to a variable 
`final paymentTypeFilters = paymentFilters.filters;`

This way, the non-null assertion operator(**!**) isn't necessary inside & after (`paymentTypeFilters != null`) condition.

Reply:
I agree and tried, but the flutter throws an `unchecked_use_of_nullable_value`

Reply:
It shouldn't, make sure you replace every instance of the variable. Most IDE's refactoring tools handle this automatically.